File: micronaut-camunda-bpm-feature/transactional-test-data-jdbc/src/test/java/info/novatec/micronaut/camunda/bpm/feature/test/PersistBookDelegate.java
package info.novatec.micronaut.camunda.bpm.feature.test;

import io.micronaut.context.annotation.Replaces;
import org.camunda.bpm.engine.delegate.DelegateExecution;
import org.camunda.bpm.engine.delegate.JavaDelegate;

import javax.inject.Named;
import javax.inject.Singleton;

/**
 * {@link JavaDelegate} which replaces the {@link DoNothingDelegate} and included the persistence of an entity.
 *
 * @author Tobias Schäfer
 */
@Singleton
@Replaces(DoNothingDelegate.class)
@Named("doSomethingDelegate")
public class PersistBookDelegate implements JavaDelegate {

    private final BookRepository bookRepository;

    public PersistBookDelegate(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    @Override
    public void execute(DelegateExecution execution) {
        bookRepository.save(new Book(execution.getBusinessKey()));
    }
}

File: micronaut-camunda-bpm-feature/transactional-test-data-jdbc/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/PersistBookDelegate.kt
package info.novatec.micronaut.camunda.bpm.feature.test

import io.micronaut.context.annotation.Replaces
import org.camunda.bpm.engine.delegate.DelegateExecution
import org.camunda.bpm.engine.delegate.JavaDelegate
import javax.inject.Named
import javax.inject.Singleton

/**
 * [JavaDelegate] which replaces the [DoNothingDelegate] and included the persistence of an entity.
 *
 * @author Tobias Schäfer
 */
@Singleton
@Replaces(DoNothingDelegate::class)
@Named("doSomethingDelegate")
class PersistBookDelegate(private val bookRepository: BookRepository) : JavaDelegate {
    override fun execute(execution: DelegateExecution) {
        bookRepository.save(Book(execution.businessKey))
    }
}